library/ssl_tls.c

```diff
@@ -1937,8 +1943,8 @@ static int ssl_parse_certificate_chain( mbedtls_ssl_context *ssl,
                 MBEDTLS_SSL_DEBUG_MSG( 1, ( "new server cert during renegotiation" ) );
                 mbedtls_ssl_send_alert_message( ssl,
                                                 MBEDTLS_SSL_ALERT_LEVEL_FATAL,
-                                                MBEDTLS_SSL_ALERT_MSG_ACCESS_DENIED );
-                return( MBEDTLS_ERR_SSL_BAD_HS_CERTIFICATE );
+                                                MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER );
```

Comment:
This alert signals a security problem. While illegal_parameter could kind of make sense since the security problem happens to be an inconsistency between two values, the more specific access_denied would make more sense. So please keep `ACCESS_DENIED`.